doc/assets/expr.dot

```diff
@@ -0,0 +1,5 @@
+graph {
```

Comment:
:+1: for the graph! Only one thing: :bulb: could the `ID_type` node been given some dots below it or something similar to indicate that it's not just a string, but a `typet`?

Reply:
Made the graph more complete.

Reply:
@peterschrammel or @smowton, could you please check for correctness?